src/Http/Http.Extensions/gen/Microsoft.AspNetCore.Http.ValidationsGenerator/Emitters/ValidationsGenerator.Emitter.cs

```diff
@@ -238,12 +240,7 @@ private static void EmitValidatableMemberForCreate(ValidatableProperty member, C
 
     private static string SanitizeTypeName(string typeName)
     {
-        // Replace invalid characters with underscores and remove generic notation
-        return typeName
-            .Replace(".", "_")
-            .Replace("<", "_")
-            .Replace(">", "_")
-            .Replace(",", "_")
-            .Replace(" ", "_");
+        // Replace invalid characters with underscores
+        return InvalidNameCharsRegex.Replace(typeName, "_");
```

Comment:
It can't start with a digit presumably?

Reply:
I don't think there's a restriction on metadata names starting with digits per the CLR spec. However, since we derive this MetadataName that we pass to this method from a TypeSymbol resolve via static analysis, it's probably come from C# code where an identifier starting with a number is invalid.

If we do hit that case, I think emitting a method name starting with `_` is OK syntax so the generator shouldn't emit invalid syntax.